openedx/core/djangoapps/notifications/forms.py

```diff
@@ -0,0 +1,64 @@
+"""
+Discussion API forms
+"""
+from django.core.exceptions import ValidationError
+from django.forms import BooleanField, CharField, Form
+
+from openedx.core.djangoapps.notifications.models import get_notification_preference_config, get_notification_channels
+
+
+class NotificationPreferenceUpdateForm(Form):
```

Comment:
Why using a form instead of a serialiser?

Reply:
+1  I was thinking the same, custom validators are good option https://www.django-rest-framework.org/api-guide/validators/#writing-custom-validators